modules/dists/SparseBlockDist.chpl

```diff
@@ -117,6 +118,64 @@ class SparseBlockDom: BaseSparseDom {
     dsiAdd((ind,));
   }
 
+  proc dsiBulkAdd(inds: [] index(rank, idxType), isSorted=false, isUnique=false,
+      preserveInds=true){
+  
+    if !isSorted && preserveInds {
+      var _inds = inds;
+      bulkAdd_help(_inds, isSorted, isUnique); 
+    }
+    else {
+      bulkAdd_help(inds, isSorted, isUnique);
+    }
+  }
+
+  proc bulkAdd_help(inds: [] index(rank,idxType), isSorted=false, isUnique=false) {
+
+    if !isSorted then
+      QuickSort(inds);
+
+    var localIndexes: [dist.targetLocDom] [inds.domain] index(rank,idxType);
+    var localIndexCnts: [dist.targetLocDom] int =
+      inds.domain.low;
+    if bulkAddMemoize {
+      var lastLocaleIndexCache = dist.targetLocDom.low;
+      for i in inds {
+        if locDoms[lastLocaleIndexCache].parentDom.member(i) {
+          localIndexes[lastLocaleIndexCache][localIndexCnts[lastLocaleIndexCache]] = i;
+          localIndexCnts[lastLocaleIndexCache] += 1;
+        }
+        else {
+          for localeIndex in dist.targetLocDom {
+            if locDoms[localeIndex].parentDom.member(i) {
+              localIndexes[localeIndex][localIndexCnts[localeIndex]] = i;
+              localIndexCnts[localeIndex] += 1;
+              lastLocaleIndexCache = localeIndex;
+              break;
+            }
+          }
+        }
+      }
+    }
+    else {
+      for i in inds {
+        for localeIndex in dist.targetLocDom {
+          if locDoms[localeIndex].parentDom.member(i) {
+            localIndexes[localeIndex][localIndexCnts[localeIndex]] = i;
+            localIndexCnts[localeIndex] += 1;
+          }
+        }
+      }
+    }
+    coforall localeIdx in dist.targetLocDom do {
+      on dist.targetLocales(localeIdx) do {
+        const locCnt = localIndexCnts[localeIdx];
+        locDoms[localeIdx].mySparseBlock.bulkAdd(localIndexes[localeIdx][0..#locCnt], 
+            true, false, false); 
```

Comment:
I think it would be nice to call with named arguments like isSorted=true
here for the rest of us that can't remember what the i'th argument corresponds to